express-mutations/server.ts

```diff
@@ -0,0 +1,121 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+// Endpoint for testing
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.body;
+    if (firstName === undefined || lastName === undefined) {
+      res.sendStatus(400);
+    }
+    const sql = `
+    insert into
+        "actors" ("firstName", "lastName")
+    values
+        ($1, $2)
+        returning *;
+    `;
+    const params = [firstName, lastName];
+    const result = await db.query(sql, params);
+    const newActor = result.rows[0];
+    res.status(201).json(newActor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.put('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    const { firstName, lastName } = req.body;
+    if (
+      firstName === undefined ||
+      lastName === undefined ||
+      !Number.isInteger(+actorId)
+    ) {
+      res.sendStatus(400);
+    }
```

Comment:
You need to return after sending the response. Otherwise the code just continues executing, which will run the SQL even though it's incorrect. You don't want that because you can corrupt your database. Also, it will then try to return a second response, which will cause the server to hang. This is why we throw a new ClientError.